public/diagnostic.js

```diff
@@ -23,3 +24,22 @@
 //  -  Order 1 : 2 hats ($5 each)
 //  -  Order 2 : 1 desk lamp ($20 each)
 //  -  Order 3 : 3 hand towels ($8 each)
+const Order = Ember.object.extend({
+  unitPrice: 0,
+  quantity: 0,
+  orderPrice: Ember.computed('unitPrice', 'quantity', function(){
+    return this.get(unitPrice) * this.get(quantity);
+  })
+});
+
+const Cart = Ember.object.extend({
+  orders: [],
+  addToCart: function(item){
+    this.orders.push(item);
+  },
```

Comment:
We need to use getters and setters to take advantage of Ember's benefits, so instead of `this.orders.push(item)` it would be  `this.get('orders').addObject(item)`